client-server-server/src/main/scala/zio/raft/server/RaftServer.scala

```diff
@@ -0,0 +1,471 @@
+package zio.raft.server
+
+import zio._
+import zio.stream._
+import zio.raft.protocol._
+import zio.zmq.RoutingId
+import scodec.bits.ByteVector
+import java.time.Instant
+
+object RaftServer {
+  
+  /**
+   * Actions initiated by the server (internal or from Raft).
+   */
+  sealed trait ServerAction
+  
+  object ServerAction {
+    case class SendResponse(sessionId: SessionId, response: ClientResponse) extends ServerAction
+    case class SendServerRequest(sessionId: SessionId, request: ServerRequest) extends ServerAction
+    case class LeadershipChange(isLeader: Boolean, sessions: Map[SessionId, SessionMetadata]) extends ServerAction
+  }
+  
+  /**
+   * Actions to forward to Raft state machine.
+   */
+  sealed trait RaftAction
+  
+  object RaftAction {
+    case class CreateSession(sessionId: SessionId, capabilities: Map[String, String]) extends RaftAction
+    case class ClientRequest(sessionId: SessionId, requestId: RequestId, payload: ByteVector) extends RaftAction
+    case class ExpireSession(sessionId: SessionId) extends RaftAction
+  }
+  
+  /**
+   * Transport abstraction for ZeroMQ SERVER socket.
+   */
+  trait ZmqServerTransport {
+    def sendMessage(routingId: RoutingId, message: ServerMessage): Task[Unit]
+    def incomingMessages: ZStream[Any, Throwable, IncomingMessage]
+  }
+  
+  case class IncomingMessage(
+    routingId: RoutingId,
+    message: ClientMessage
+  )
+  
+  case class SessionConnection(
+    routingId: Option[RoutingId],
+    expiresAt: Instant
+  )
+  
+  case class SessionMetadata(
+    capabilities: Map[String, String],
+    createdAt: Instant
+  )
+  
+  /**
+   * Reference for generating unique request IDs.
```

Comment:
server is not generating request ids, the raft state machine generate them for each ServerRequest.